crud/select/executor.lua

```diff
@@ -53,10 +57,10 @@ function executor.execute(space, index, filter_func, opts)
         if value == nil then
             value = opts.after_tuple
         else
-            local cmp_operator = select_comparators.get_cmp_operator(opts.tarantool_iter)
-            local scan_comparator = select_comparators.gen_tuples_comparator(cmp_operator, index.parts)
-            local after_tuple_key = utils.extract_key(opts.after_tuple, index.parts)
-            if scan_comparator(after_tuple_key, opts.scan_value) then
+            local key_def = keydef_lib.new(index.parts)
+            local after_tuple_key = key_def:extract_key(opts.after_tuple)
```

Comment:
Please move this line under `key_def:compare_with_key(opts.after_tuple, opts.scan_value)` - you don't need to extract anything if condition below will be false.

 Also I `== 0` means "equal" I but `scan_comparator(after_tuple_key, opts.scan_value)` returned `true` in case of "less than". So I'm not sure that your changes are correct.